src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,264 @@
+    #  - i think i want to show instances where there is an obsolete source in the evidence...
+    #  - but i don't want to sync it
+    #  - i guess it's alright to remove them because this will only remove rows where 100% of its mappings are obsolete.
+    #  - so the obsolete ID should still show up in mondo_df
+    #    - test this?
+    # - filter by source previously filtered mappings_df (obsoletes & deletions)
+    mondo_df = mondo_df[mondo_df['mondo_id'].isin(set(mappings_df['mondo_id']))]  # alternative: JOIN-based solution
+    # todo: filter mappings? how? (yes?)
+    #   - i think yes, because otherwise other sources will show up in results
+    #   todo: should mondo_df len should be about same for all sources? (no?)
+    #    - I think No, because I have already added evidence before filter. so prob don't need it now
+    # - filter out rows which don't appear in mondo.sssom.tsv for whatever reason
+    mondo_df = mondo_df.merge(mappings_df[['mondo_id', 'source_id']], on=['mondo_id', 'source_id'], how='inner')
+    # todo: filter NON_SOURCE_ID_SOURCE_SPECIFIC_SYNONYM_XREFS? (yes?)
+    #   Do I even need to filter these out now?
+    #   This mattered when I was JOINing on source_id. but now, i want to list all source evidence in a new col.
+    #   might as well have this in there
+    #   - probalby yes again because i've already captured the evidence
+    # - filter out any non-source-ID, source-specific synonym xrefs
+    #   Not future-proof, as this list has to be maintained, but as of 2024/07/15 there is only one entry, so there is
+    #   not high risk of this being a problem. Alternative: Come up w/ regex for valid codes for all sources. Will
+    #   mostly be numbers, but can include . and -. Not sure if can include alpha chars.
+    mondo_df = mondo_df[~mondo_df['source_id'].isin(NON_SOURCE_ID_SOURCE_SPECIFIC_SYNONYM_XREFS)]
+    # - add labels
+    mondo_df = mondo_df.merge(mappings_df, on=['mondo_id', 'source_id'], how='left')
+
+    # Determine synchronization cases    # -confirmed
+    confirmed_df = mondo_df.merge(source_df, on=['synonym_scope', 'synonym'], how='inner').rename(columns={
+        'source_id_x': 'source_id'})
+    del confirmed_df['source_id_y']
+    _format_and_save(confirmed_df, outpath_confirmed)
+    confirmed_df['case'] = 'confirmed'
+
+    # -updated
+    updated_df = mondo_df.merge(source_df, on=['synonym'], how='inner').rename(columns={
+        'synonym_scope_x': 'synonym_scope_mondo', 'synonym_scope_y': 'synonym_scope', 'source_id_x': 'source_id' })
+    del updated_df['source_id_y']
+    updated_df = updated_df[updated_df['synonym_scope_mondo'] != updated_df['synonym_scope']]
+    updated_df = updated_df[[
+        'mondo_id', 'mondo_label', 'source_id', 'source_label', 'synonym_scope_mondo', 'synonym_scope', 'synonym',
+        'synonym_type']]
+    _format_and_save(updated_df, outpath_updated)
+    updated_df['case'] = 'updated'
+
+    # -added
+    added_df = source_df.merge(
```

Comment:
## Bug?: `-added`: synonym is Mondo term's label
Things are getting added to the `-added` template even when the synonym from the source exists in Mondo not as a synonym, but as the label.

**Example:**
Mondo
```owl
<owl:Class rdf:about="http://purl.obolibrary.org/obo/MONDO_8000008">
    <rdfs:label>Martsolf syndrome 1</rdfs:label>
```

OMIM
```
<owl:Class rdf:about="https://omim.org/entry/212720">
    <oboInOwl:hasExactSynonym>martsolf syndrome 1</oboInOwl:hasExactSynonym>
    <rdfs:label>martsolf syndrome 1</rdfs:label>
```

The capitalization differs, but that doesn't matter.
Also I thought all Mondo labels were supposed to be lowercase?

FYI the only reason this one is a synonym in the first place is because in mondo-ingest during component build, we make synonyms out of all labels.

Reply:
> Things are getting added to the -added template even when the synonym from the source exists in Mondo not as a synonym, but as the label.

This is ok! Labels and Synonyms can be the same / redundant, no problem.

> The capitalization differs, but that doesn't matter.

`Martsolf` must be a name of a person, that is why it is capital.

> I think this is a bug, but will clarify with Nico.

No its fine, you can add synonyms to the "added" table if they correspond to existing labels.

Reply:
@twhetzel FYI you were right; non issue. Can mark resolved.